x/gmp/module_ibc.go

```diff
@@ -0,0 +1,211 @@
+	return IBCModule{
+		app: app,
+	}
+}
+
+// OnChanOpenInit implements the IBCModule interface
+func (im IBCModule) OnChanOpenInit(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID string,
+	channelID string,
+	chanCap *capabilitytypes.Capability,
+	counterparty channeltypes.Counterparty,
+	version string,
+) (string, error) {
+	return im.app.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, version)
+}
+
+// OnChanOpenTry implements the IBCModule interface
+func (im IBCModule) OnChanOpenTry(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID,
+	channelID string,
+	chanCap *capabilitytypes.Capability,
+	counterparty channeltypes.Counterparty,
+	counterpartyVersion string,
+) (string, error) {
+	return im.app.OnChanOpenTry(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, counterpartyVersion)
+}
+
+// OnChanOpenAck implements the IBCModule interface
+func (im IBCModule) OnChanOpenAck(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyChannelID string,
+	counterpartyVersion string,
+) error {
+	return im.app.OnChanOpenAck(ctx, portID, channelID, counterpartyChannelID, counterpartyVersion)
+}
+
+// OnChanOpenConfirm implements the IBCModule interface
+func (im IBCModule) OnChanOpenConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	return im.app.OnChanOpenConfirm(ctx, portID, channelID)
+}
+
+// OnChanCloseInit implements the IBCModule interface
+func (im IBCModule) OnChanCloseInit(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	return im.app.OnChanCloseInit(ctx, portID, channelID)
+}
+
+// OnChanCloseConfirm implements the IBCModule interface
+func (im IBCModule) OnChanCloseConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	return im.app.OnChanCloseConfirm(ctx, portID, channelID)
+}
+
+// OnRecvPacket implements the IBCModule interface
+func (im IBCModule) OnRecvPacket(
+	ctx sdk.Context,
+	modulePacket channeltypes.Packet,
+	relayer sdk.AccAddress,
+) ibcexported.Acknowledgement {
+	// this line is used by starport scaffolding # oracle/packet/module/recv
+
+	var data transfertypes.FungibleTokenPacketData
+	var err error
+	if err := types.ModuleCdc.UnmarshalJSON(modulePacket.GetData(), &data); err != nil {
+		ctx.Logger().Debug(fmt.Sprintf("cannot unmarshal ICS-20 transfer packet data: %s", err.Error()))
+		return im.app.OnRecvPacket(ctx, modulePacket, relayer)
+	}
+
+	var msg Message
+	if err = json.Unmarshal([]byte(data.GetMemo()), &msg); err != nil {
+		ctx.Logger().Debug(fmt.Sprintf("cannot unmarshal memo: %s", err.Error()))
+		return im.app.OnRecvPacket(ctx, modulePacket, relayer)
+	}
+
+	if msg.Payload == nil {
+		return im.app.OnRecvPacket(ctx, modulePacket, relayer)
+	}
+
+	// // authenticate the message with packet sender + channel-id
+	// // TODO: authenticate the message with channel-id
+	// if data.Sender != AxelarGMPAcc {
+	// 	return ack
+	// }
```

Comment:
@luckychess, r.e. the TODO above, we are not introducing any new privileged execution or security assumptions to the transfer chain with this module. We should be safe to passthrough without a sender, port, channel condition.

Reply:
I agree, it was a suggestion from the example integration code I followed to.